packages/react-storage/src/components/StorageBrowser/controls/hooks/useNavigation.ts

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { NavigationProps } from '../../composables/Navigation';
+import { useControlsContext } from '../../controls/context';
+
+export const useNavigation = (): NavigationProps | null => {
+  const { data, onNavigate, onNavigateHome } = useControlsContext();
+  const { location } = data;
+
+  const { current, path = '' } = location ?? {};
+
+  return React.useMemo(() => {
+    if (!current) {
+      return null;
```

Comment:
Can we return an empty breadcrumb or maybe `[{ name: 'Home', onNavigate: onNavigateHome }]`  as on line 35? consumers of the hook could then avoid null checks.